extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -47,6 +50,40 @@ export async function findLanguage(
   return language;
 }
 
+interface RepoListQuickPickItem extends QuickPickItem {
+  repoList: string[];
+}
+
+/**
+ * Gets the repositories to run the query against.
+ */
+async function getRepositories(): Promise<string[] | undefined> {
+  if (REPO_LISTS) {
+    const quickPickItems = Object.keys(REPO_LISTS).map<RepoListQuickPickItem>(key => (
+      {
+        label: key,                // the name of the repository list
+        repoList: REPO_LISTS[key], // the actual array of repositories
+      }
+    ));
```

Comment:
nit: Maybe a little cleaner.

```suggestion
    const quickPickItems = Object.enties(REPO_LISTS).map<RepoListQuickPickItem>(([key, value]) => (
      {
        label: key,                // the name of the repository list
        repoList: value, // the actual array of repositories
      }
    ));
```

Reply:
Also, note that repositories lists is a user setting and it may be malformed. Just make sure that if a malformed list (eg- a single string or even a number) is passed, then the error is expressed to the user in an understandable way.

Reply:
> Also, note that repositories lists is a user setting and it may be malformed. Just make sure that if a malformed list (eg- a single string or even a number) is passed, then the error is expressed to the user in an understandable way.

👍🏽 Good point. I've re-instated the [package.json](https://github.com/github/vscode-codeql/pull/918/files#diff-be4c5a31505df7de81fa4900e84e66c04d1a6bca9b3da50f8061d2cb2bfbbde1) change, which at least validates that each list is an array of strings. (I removed it from the package.json initially since that stops it from showing up in the settings UI, but maybe we don't need to be too secretive 🤫 )